### PR TITLE
Add 4 button aqara opple switch to supported devices

### PR DESCRIPTION
**Pull request by tr1ng0**
Try No. 2! 
Alright so my first approch was not successfull due the fact that Xiaomi-Wireless-Switch only maps 1 button ... well here I go 

This code is only for the 4 Button Version:
https://www.zigbee2mqtt.io/devices/WXCJKG12LM.html

The Wireless Switch was tested so it works (at least on my end) and is already configuered into my Homekit.

I will add support for the 2 and 6 Button Version shortly after this PR is merged because there are a few technical questions open: handling different version (2,4,6 buttons) of the same type of Wireless Switch (maybe someone has an idea for that)
 
There are 2 Questions for @madchicken:
I personally start counting from top left to bottom right. I don't know why but currently the Buttons are configuered like that:
top Left Button: Button 3 in Home App
top Right Button: Button 4 in Home App
bottom Left Button: Button 1 in Home App
bottom Right Button: Button 2 in Home App
I personally don't care because Home App shows the button u just pressed while editing the different actions in home app. What is your opinion here and/or do you know what I did wrong?

Second: 
The Wireless Switch supports tripple_press which led to errors while compiling (don't ask me why its midnight pls forgive me). So I commented the section out handling the tripple_press i mean who on earth uses tripple press? With Single,Double, Long u already have 12 different triggers the heck u want with 16 (and who remebers that all of those?!)

Well it is still my first contribution to an open-source Project! Constructive Feedback is really welcomed! 

Regards,
tr1ng0

### CODE REVIEW

File: src/accessories/xiaomi/aqara-opple-switch.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+        this.switchServiceBottomLeft, 
+        this.switchServiceBottomRight,
+        this.switchServiceTopLeft,
+        this.switchServiceTopRight,
+        this.batteryService,
+    ] = builder
+      .withStatelessSwitch('button_1', 'top_left', 1, [
+        ProgrammableSwitchEvent.SINGLE_PRESS,
+        ProgrammableSwitchEvent.DOUBLE_PRESS,
+        //ProgrammableSwitchEvent.TRIPPLE_PRESS,
```
Comment (madchicken) [inline src/accessories/xiaomi/aqara-opple-switch.ts:33]: HomeKit does not support triple press, just remove this

File: src/accessories/xiaomi/aqara-opple-switch.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+        this.switchServiceBottomLeft
+          .getCharacteristic(ProgrammableSwitchEvent)
+          .setValue(ProgrammableSwitchEvent.DOUBLE_PRESS);
+        break;
+      case 'button_4_double':
+        this.switchServiceBottomRight
+          .getCharacteristic(ProgrammableSwitchEvent)
+          .setValue(ProgrammableSwitchEvent.DOUBLE_PRESS);
+        break;
+    // tripple press
```
Comment (madchicken) [inline src/accessories/xiaomi/aqara-opple-switch.ts:112]: Same as above (not supported)

**Review by madchicken — CHANGES_REQUESTED** [review]
This is good, just remove commented code please.

**Comment by madchicken** [comment]
> There are 2 Questions for @madchicken:
> I personally start counting from top left to bottom right. I don't know why but currently the Buttons are configuered like that:
> top Left Button: Button 3 in Home App
> top Right Button: Button 4 in Home App
> bottom Left Button: Button 1 in Home App
> bottom Right Button: Button 2 in Home App
> I personally don't care because Home App shows the button u just pressed while editing the different actions in home app. What is your opinion here and/or do you know what I did wrong?

No, nothing wrong. It really depends on the device an probably on the zigbee lib and how it maps it. I would suggest mapping buttons in the correct order in the `getAvailableServices` function so they appear in the "proper" order also in the Home App (in your case 3-4-1-2).

**Comment by madchicken** [comment]
Question for you @tr1ng0 : I see [WXCJKG13LM](https://www.zigbee2mqtt.io/devices/WXCJKG13LM.html) model has 6 buttons. Wouldn't it be possible to map both devices in the same class (by just adding an if on the modelId)?

**Comment by tr1ng0** [comment]
TRIPPLE_PRESS is now fully removed from the PR 
I started adding some code for the 6 Button version, not implemented yet
changing the index of the buttons resolved in unable to double,long press, I'll need to investigate further because this could have something todo with my homebridge being broken :( 

> Question for you @tr1ng0 : I see [WXCJKG13LM](https://www.zigbee2mqtt.io/devices/WXCJKG13LM.html) model has 6 buttons. Wouldn't it be possible to map both devices in the same class (by just adding an if on the modelId)?
A: Yes thats exactly what I thought, I'll get to it later

**Comment by tr1ng0** [comment]
Bildschirmfoto 2021-01-12 um 12 41 20
Is this normal behaviour? 

Bildschirmfoto 2021-01-12 um 12 42 49
I'm somehow unable unpair the device ... are there any additional setup I have to do? @madchicken 
Could this "failed" unpairing be the reason for my Port 9000 Error: Cannot GET /

**Comment by madchicken** [comment]
That's strange. Which version of the plugin are you using? When the UI comes up you should see in the logs (something like `Listening on ...`)

**Comment by tr1ng0** [comment]
> That's strange. Which version of the plugin are you using? When the UI comes up you should see in the logs (something like `Listening on ...`)

To what is this refering? 
Well I'm not sure ... I just change the whole dist file i compiled with the old one in the node_modules section

**Comment by madchicken** [comment]
> To what is this refering?
> Well I'm not sure ... I just change the whole dist file i compiled with the old one in the node_modules section

The web UI has an internal web server that is started when the plugin/homebridge starts (and that line of log is dropped). The fact that you are not able to unpair the device should not be related to the fact that you are not able to reach the UI.
To remove your device you can also try to manually remove it from the database (it's under '/var/local/lib/homebridge/zigbee.db') and then remove it from homebridge caches (you can do that by accessing the `/settings` page of homebrige UI)

**Comment by tr1ng0** [comment]
@madchicken thank you that manual removal worked wonders! Buttons are now correctly set up and match expected behaviour inside home app

Is it still normal that a switch shows all diferent buttons? This is the only wireless switch i own

**Comment by madchicken** [comment]
> Is it still normal that a switch shows all diferent buttons? This is the only wireless switch i own

Yes it is. But in your Home App it should appear as a single device that has multiple buttons you can configure

**Review by madchicken — APPROVED** [review]

**Comment by tr1ng0** [comment]
@madchicken yes it appears as one! Thank you for your support

**Pull request merged**